extensions/amp-story/1.0/analytics.js

```diff
@@ -70,6 +79,13 @@ export class AmpStoryAnalytics {
     this.triggerEvent_(event);
   }
 
+  /**
+   * @param {AdvancementMode} mode
+   */
+  onAdvancementModeStateChange(mode) {
```

Comment:
I think you can remove this (and the call to it on line 546 of `amp-story.js`), since we don't want to actually trigger a new analytics ping when the advancement mode changes.  Updating the variable service should be sufficient; then when the next analytics trigger fires (e.g. when the story changes pages, or when the user unmutes the story, or when the user hits the bookend, etc), the variables will be populated because of the rest of what you've done here.

Reply:
Thanks for taking the time to explain this. Makes a lot of sense, and removed.